src/EditorFeatures/Core/DocumentationComments/DocumentationCommentSuggestion.cs

```diff
@@ -65,26 +65,39 @@ public override Task OnProposalUpdatedAsync(SuggestionSessionBase session, Propo
             return Task.CompletedTask;
         }
 
-        public async Task TryDisplaySuggestionAsync(CancellationToken cancellationToken)
+        public async Task<bool> StartSuggestionSessionAsync(CancellationToken cancellationToken)
         {
-            _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false);
+            await RunWithEnqueueActionAsync(
+                "StartWork",
+                async () => _suggestionSession = await SuggestionManager.TryDisplaySuggestionAsync(this, cancellationToken).ConfigureAwait(false),
+            cancellationToken).ConfigureAwait(false);
 
-            if (_suggestionSession != null)
+            if (_suggestionSession is null)
             {
-                var success = await TryDisplayProposalAsync(_suggestionSession, cancellationToken).ConfigureAwait(false);
-                if (success)
-                {
-                    Logger.Log(FunctionId.Copilot_Generate_Documentation_Displayed, logLevel: LogLevel.Information);
-                }
+                await DisposeAsync().ConfigureAwait(false);
+                return false;
+            }
+
+            return true;
+        }
+
+        public async Task TryDisplayDocumentationSuggestionAsync(ProposalBase proposal, CancellationToken cancellationToken)
+        {
+            var success = await TryDisplayProposalAsync(proposal, cancellationToken).ConfigureAwait(false);
+            if (success)
+            {
+                Logger.Log(FunctionId.Copilot_Generate_Documentation_Displayed, logLevel: LogLevel.Information);
             }
         }
```

Comment:
is it needed to spearate out the concept of 'start' and 'try display'?  why not just merge these so that the caller just tries to display things, and this moves us into the start mode?